### PR TITLE
fix some bugprone-macro-parentheses warnings

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
## Description
When using the public headers of catch2 in another project that uses clang-tidy with some checks enabled, then some warnings in catch2's headers are also reported. Those can distract from the real problem in our project. So I would like to reduce the warnings from catch2.

This fixes a bunch of bugprone-macro-parentheses warnings.

See: https://clang.llvm.org/extra/clang-tidy/checks/bugprone/macro-parentheses.html

## GitHub Issues
None

### CODE REVIEW

**Comment by horenmar** [comment]
This does not even compile.

**Comment by ChristianTackeGSI** [comment]
Okay, in my first attempt this used to compile. Now it also fails.

Anyway: I have stripped this down to the part that does compile and also gives a successful `make test`. Hope this is better now?

**Comment by horenmar** [comment]
Thanks, I'll take a look.

> Okay, in my first attempt this used to compile. Now it also fails.

I suspect that you didn't have test build enabled, and because the main library doesn't exercise those macros, breaking them won't stop it from compiling.

**Comment by ChristianTackeGSI** [comment]
Is there anything left, that I forgot about?
What's the next step?

**Comment by horenmar** [comment]
You haven't forgotten about anything, I did forget about the PR though.

File: src/catch2/catch_message.hpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ namespace Catch {
 
 ///////////////////////////////////////////////////////////////////////////////
 #define INTERNAL_CATCH_UNSCOPED_INFO( macroName, log ) \
-    Catch::getResultCapture().emplaceUnscopedMessage( Catch::MessageBuilder( macroName##_catch_sr, CATCH_INTERNAL_LINEINFO, Catch::ResultWas::Info ) << log )
```
Comment (horenmar) [inline src/catch2/catch_message.hpp]: This breaks streaming usage like

`UNSCOPED_INFO("Status is " << status.ok());`

I see that we do not have this test for `UNSCOPED_INFO`, only for `INFO` -> I assume that's why this change isn't on the `INFO` internals as well.

I added the test in 9c541ca7

**Review by horenmar — COMMENTED** [review]
I am willing to merge the other changes, as they should not break anything.

File: src/catch2/catch_message.hpp (hunk 1 above)
Comment (ChristianTackeGSI) [reply src/catch2/catch_message.hpp]: Okay, removed that part.
